app-frontend/src/app/redux/node-utils.js

```diff
@@ -5,9 +5,7 @@ import {authedRequest} from '../api/authentication';
 import {colorStopsToRange} from '_redux/histogram-utils';
 import {createRenderDefinition} from '_redux/histogram-utils';
 
-export function astFromNodes(labState, updatedNodes) {
-    const analysis = labState.analysis;
-    const nodes = labState.nodes;
+export function astFromNodesStateless(analysis, nodes, updatedNodes) {
```

Comment:
I feel a little bit strange about the naming of this - these functions are all pure, but one of them takes a state object as a function, the other takes separates the relevant stuff out into separate function parameters...

Reply:
I think it might make more sense just to avoid creating a separate function to handle them separated out and make the "state" object you pass in on the fly where it gets used.

Looking back, as a general pattern, it would probably make more sense to make the signature something like this:
`export function astFromNodes({analysis, nodes}, updateNodes)` and avoid mentioning state at all, 
or just separating the parameters out on all of them and not taking a state object in the first place. I don't think it's worth walking back and refactoring this stuff at the point though.